lib/src/main/java/uk/gov/di/ipv/core/library/service/ClientOAuthSessionDetailsService.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.di.ipv.core.library.service;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
+import uk.gov.di.ipv.core.library.helpers.SecureTokenHelper;
+import uk.gov.di.ipv.core.library.persistence.DataStore;
+import uk.gov.di.ipv.core.library.persistence.item.ClientOAuthSessionItem;
+
+import java.text.ParseException;
+
+import static uk.gov.di.ipv.core.library.config.ConfigurationVariable.BACKEND_SESSION_TTL;
+import static uk.gov.di.ipv.core.library.config.EnvironmentVariable.CLIENT_OAUTH_SESSIONS_TABLE_NAME;
+
+public class ClientOAuthSessionDetailsService {
+    private final DataStore<ClientOAuthSessionItem> dataStore;
+    private final ConfigService configService;
+
+    @ExcludeFromGeneratedCoverageReport
+    public ClientOAuthSessionDetailsService(ConfigService configService) {
+        this.configService = configService;
+        boolean isRunningLocally = this.configService.isRunningLocally();
+        dataStore =
+                new DataStore<>(
+                        this.configService.getEnvironmentVariable(CLIENT_OAUTH_SESSIONS_TABLE_NAME),
+                        ClientOAuthSessionItem.class,
+                        DataStore.getClient(isRunningLocally),
+                        isRunningLocally,
+                        configService);
+    }
+
+    public ClientOAuthSessionDetailsService(
+            DataStore<ClientOAuthSessionItem> dataStore, ConfigService configService) {
+        this.dataStore = dataStore;
+        this.configService = configService;
+    }
+
+    public ClientOAuthSessionItem getClientOAuthSession(String clientOAuthSessionId) {
+        return dataStore.getItem(clientOAuthSessionId);
+    }
+
+    public ClientOAuthSessionItem generateClientSessionDetails(
+            JWTClaimsSet claimsSet, String clientId) throws ParseException {
+        ClientOAuthSessionItem clientOAuthSessionItem = new ClientOAuthSessionItem();
+
+        clientOAuthSessionItem.setClientOAuthSessionId(SecureTokenHelper.generate());
+        clientOAuthSessionItem.setResponseType(claimsSet.getStringClaim("response_type"));
+        clientOAuthSessionItem.setClientId(clientId);
+        clientOAuthSessionItem.setRedirectUri(claimsSet.getStringClaim("redirect_uri"));
+        clientOAuthSessionItem.setState(claimsSet.getStringClaim("state"));
+        clientOAuthSessionItem.setUserId(claimsSet.getSubject());
+        clientOAuthSessionItem.setGovukSigninJourneyId("govuk_signin_journey_id");
```

Comment:
I think, should read that from claimsSet.getStringClaim("govuk_signin_journey_id") like generateClientSessionDetails()

Reply:
@TalatAydinGDS have changed as you've suggested & updated tests